test-framework/junit5-vertx/src/main/java/io/quarkus/test/junit/vertx/RunOnVertxContextTestMethodInvoker.java

```diff
@@ -0,0 +1,129 @@
+        }
+        uniAsserter = new DefaultUniAsserter();
+        return uniAsserter;
+    }
+
+    @Override
+    public boolean supportsMethod(Class<?> originalTestClass, Method originalTestMethod) {
+        return hasAnnotation(RunOnVertxContext.class, originalTestMethod.getAnnotations())
+                || hasAnnotation(RunOnVertxContext.class, originalTestClass.getAnnotations());
+    }
+
+    // we need to use the class name to avoid ClassLoader issues
+    private boolean hasAnnotation(Class<? extends Annotation> annotation, Annotation[] annotations) {
+        if (annotations != null) {
+            for (Annotation methodAnnotation : annotations) {
+                if (annotation.getName().equals(methodAnnotation.annotationType().getName())) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public Object invoke(Object actualTestInstance, Method actualTestMethod, List<Object> actualTestMethodArgs,
+            String testClassName) throws Throwable {
+
+        Vertx vertx = VertxCoreRecorder.getVertx().get();
+        if (vertx == null) {
+            throw new IllegalStateException("Vert.x instance has not been created before attempting to run test method '"
+                    + actualTestMethod.getName() + "' of test class '" + testClassName + "'");
+        }
+        CompletableFuture<Object> cf = new CompletableFuture<>();
+        RunTestMethodOnContextHandler handler = new RunTestMethodOnContextHandler(actualTestInstance, actualTestMethod,
+                actualTestMethodArgs, uniAsserter, cf);
+        vertx.getOrCreateContext().runOnContext(handler);
+        try {
+            return cf.get();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            return null;
+        } catch (ExecutionException e) {
+            // the test itself threw an exception
+            throw e.getCause();
+        }
+    }
+
+    public static class RunTestMethodOnContextHandler implements Handler<Void> {
+        private final Object testInstance;
+        private final Method targetMethod;
+        private final List<Object> methodArgs;
+        private final DefaultUniAsserter uniAsserter;
+        private final CompletableFuture<Object> future;
+
+        public RunTestMethodOnContextHandler(Object testInstance, Method targetMethod, List<Object> methodArgs,
+                DefaultUniAsserter uniAsserter, CompletableFuture<Object> future) {
+            this.testInstance = testInstance;
+            this.future = future;
+            this.targetMethod = targetMethod;
+            this.methodArgs = methodArgs;
+            this.uniAsserter = uniAsserter;
+        }
+
+        @Override
+        public void handle(Void event) {
+            ManagedContext requestContext = Arc.container().requestContext();
+            if (requestContext.isActive()) {
```

Comment:
Isn't this fishy? AFAIK the request context is always active on the test methods, no? If so, it should just be propagated, no?

Reply:
It's actually needed. Without it I am getting ContextNotActiveException in the Panache tests